new chaplain sect, the Ever-Burning Candle sect

Caldony opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Adds the Ever-Burning Candle sect, a sect that is about candles, and also fire, mainly fire.
The Ever-Burning Candle sect has 3 rites, but first let's talk about the most important one which is candle fuel

1. **Candle Fuel:**
Candle Fuel is a rite that consist of sacrificing a burning carbon corpse buckled to the altar for favor. the more burn damage the corpse has the more favor you'll get, although most human corpses will cap at around 600 burn damage, meaning that you'll generally get a maximum of 700 favor since the rite gives you 100 favor + the burn damage of the corpse. 
Now you may think this encourage people to put every corpse in medbay on fire, but it's actually very inefficient to do that since you can use monkeys for the sacrifices, it's supposed to encourage the chaplain to work with other departments for the sacrifices may it be botany, genetics or even medical if they have suicided corpses. And in exchange the chaplain will use the one rite that matters with this sect to help others, Sacrificing a corpse that can still be revived with this rite without a good reason is still antagging as it is dusted and can never come back and as such should be treated like a chaplain cremating someone for all intents and purposes.
It's REALLY easy to get the maximum amount of burn damage by just going to lavaland and dragging them back and forth in lava, or even just waiting for them to be crispy enough for a sacrifice by just setting them on fire and waiting for a minute, and as such burn damage is rarely a problem to max out, just keep a health analyzer to know when they are hot enough.

2. **Unmelting Wax:**
Makes a piece of clothing fireproof, that's it.
Now yes the fireproof potion from xenobio already does that, for less of a cost, with two charges. But xenobio is a balance mess and it's a gamble if they will ever help you for anything. While the chaplain is often wandering around the station asking people to join their religion or sell candy or whatever holy people do. 
This rite cost 1000 favor, aka : nearly two fully burned corpse. it only fireproofs ONE piece of clothing, meaning that for full fire protection you'll need to do the rite twice on both a hood and a jumpsuit/exosuit. unlike the fireproof potion the chaplain is the only one able to do the rite, so if he want he can only fireproof "holy" clothing so that only his followers get the "blessing"

3. **Immortal Candles:**
Immortal Candles spawn 5 candles that never melt
That's it there's no powergaming or big reveal that's all they do. It just wouldn't feel right to have a candle sect with at least SOME candles in there, and we already have infinite candles in game so eeeeeh. They are mainly here for fluff or decoration since it's pretty annoying to get a whole holy candle thing going only for them to run out after 5 minutes. The rite cost 200 favor, aka 1/3 of a burning corpse, burning corpse are the currency here.

not the only one actually, you _can_ sacrifice lit candles for 20 favor, which is jack shit, but hey you have a vegan option at least.

this sect does not have any blessing with the bible, since it already has the rites to make it useful. And keeping the default healing bless takes away the one thing normalcy is good at. Also the sect_bless proc is currently broken and only works half the time even for technophiles.

**_tldr : burn corpses > get favor > get fireproof clothing_**
## Why It's Good For The Game

We both need more sects and more reason for the chaplain to interact with the rest of the station that isn't just validing magic things. I do get that the chaplain is supposed to be an RP role, but we need to mix RP with mechanics for it to work, and giving the chaplain a reason to make weird burning monkey sacrifice makes it feel more natural and actually worth it since it mechanically does something. it also lets the chaplain put their own twist on why they are doing it and maybe invite people to their religion for religion perks (fireproof clothing).

## Changelog
:cl:
add: The sect of the Ever-Burning Candle has been officially recognised by NT and can now be practiced legally in the chapel.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Caldony committed:
Adds the sect of the Ever Burning candle

Caldony committed:
make the burning sacrifice 20 seconds instead of 10

only made it 10 seconds for testing purposes, also make the "unmelting" clothing red instead of blue to be more red candle-ish

davethwave commented:
Something that I could recommend for the bible healing, is to heal only burn damage or convert brute to burn at a ratio like 5 brute = 3 burn.

ATH1909 commented:
l'amp

Caldony committed:
make another check at the end of the candle fuel rite

The candle fuel rite will now check if the sacrifice is still on fire and dead at the end of the rite, you can technically extinguish them and only light them on fire at the last moment but it's fine god won't notice

optimumtact commented:
@ExcessiveUseOfCobblestone I hate u ~

Winterous commented:
One question and one request.

Question: On things like hardsuits, are you able to bless the helmet as well, considering that it's not an item you can actually GET? Will blessing the body by extension bless the head, or would you have to do them separately, by wearing it and smacking your own helmet?

Request: Please don't give a stupid colour to blessed clothes like with the fireproofing clothing, make it more subtle, it looks HORRIBLE.

Caldony commented:
wait fuck ignore that that was an accident

Caldony commented:
> 
> 
> One question and one request.
> 
> Question: On things like hardsuits, are you able to bless the helmet as well, considering that it's not an item you can actually GET? Will blessing the body by extension bless the head, or would you have to do them separately, by wearing it and smacking your own helmet?
> 
> Request: Please don't give a stupid colour to blessed clothes like with the fireproofing clothing, make it more subtle, it looks HORRIBLE.

I hadn't thought of that actually, especially since a lot of religious garb have hoods included inside, I'm not sure if I could make it so it checks for intergated headgear or something.
as for the color I originally made it red instead of blue, but making it less glaring could be a good idea, I'll work on it

Winterous commented:
How do you actually apply the blessing, do you just smack an item with your bible after selecting it? If so you can always just equip the item and then smack the hood.

Otherwise yeah, you'll have to work out a way to include hoods in the blessing.

Caldony committed:
make the unmelting wax rite actually fireproof clothing hood/helmets

so yes you can supposedly get 2 fireproof for the price of one, but since hardsuit and hoodie are allegedly just one big connected suit it sort of make sense.
I also took away the forced color from the fireproofed clothing as making it mesh well with most clothing ended badly so now you can only tell its unmelting by the name.

Caldony committed:
few improvements to rites

makes a tighter if check for the unmelting was helmet thing. adds a sound effect at the end of each rite, gives the sect a max favor of 10 000 instead of 1000 which I could've sworn was originally the case but apparently not.

Caldony committed:
change the for loop

Caldony commented on code/modules/religion/rites.dm:
```diff
@@ -102,3 +102,116 @@
+
+/datum/religion_rites/fireproof/invoke_effect(mob/living/user, atom/religious_tool)
+	for(chosen_clothing in get_turf(religious_tool))
+		if(istype(chosen_clothing,/obj/item/clothing/suit/hooded) || istype(chosen_clothing,/obj/item/clothing/suit/space/hardsuit ))
```
yup, I could technically delete this ifcheck since no clothing that can hold items can hold head clothing, but I'd rather not take the risk.

ExcessiveUseOfCobblestone commented on code/modules/religion/rites.dm:
```diff
@@ -102,3 +102,116 @@
+	return FALSE
+
+/datum/religion_rites/fireproof/invoke_effect(mob/living/user, atom/religious_tool)
+	for(chosen_clothing in get_turf(religious_tool))
```
can you explain what you are trying to do here? After rereading this it doesnt make sense.

Chosen clothing by this point should be a singular item you have saved during perform rite, you shouldnt have to loop through anything.

you might be looking for something like `(!QDELETING(chosen_clothing) && get_turf(religious_tool) == chosen_clothing.loc) //we use loc on the second check since we want to make sure its out in the open on the tool aka altar still`

ExcessiveUseOfCobblestone commented on code/modules/religion/rites.dm:
```diff
@@ -102,3 +102,116 @@
+	if(!LAZYLEN(movable_reltool.buckled_mobs))
+		to_chat(user, "<span class='warning'>Nothing is buckled to the altar!</span>")
+		return FALSE
+	for(var/mob/living/carbon/corpse in movable_reltool.buckled_mobs) //works with any carbon corpse,not just humans
+		if(corpse.stat != DEAD)
+			to_chat(user, "<span class='warning'>You can only sacrifice dead bodies, this one is still alive!</span>")
+			return FALSE
+		if(!corpse.on_fire)
+			to_chat(user, "<span class='warning'>This corpse needs to be on fire to be sacrificed!</span>")
+			return FALSE
+		chosen_sacrifice = corpse
+		return ..()
```
if you just made it any living mob (the type) you wouldnt have to type the loop. There also doesnt seem to be any feedback that the mob needs to be a carbon.

Caldony replied:
I will have to disagree on this one, see if I don't check what is buckled with the for loop, then I can't confirm that it's a carbon, then I can't check for the other condition, so someone could buckle the corpse of a random mob, start the ritual, then swap that in with another corpse at the last moment. Now this is very unlikely to happen, but considering this is a free rite that dusts the sacrifice, I'd rather the sacrifice is done fair and square. And it's just a potential issue report waiting to happen anyway

I will however add a carbon check since it only make sense

ExcessiveUseOfCobblestone replied:
I think you misunderstood what I wanted or I wrote it shitty

https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#istypeless-for-loops

the only thing that should be in this list is living mobs, so it doesnt make sense to type it unless you are REALLY deadset on keeping it carbon.

Caldony replied:
okay done, I think I got confused by "didn't have to type the loop" so I just thought I had to remove the for loop entirely, should be fine now

ExcessiveUseOfCobblestone commented:
ping me in discord after u completed the reviews, ive been bad about checking github

Caldony committed:
Do the thing that cobby asks

I still kept the for loop for the candle fuel for reasons cited above. In other news I just learned that the word SPAM came from a Monty python skit

Caldony committed:
WHY CAN IAN BE SET ON FIRE AAAAA

Caldony committed:
what if we kissed on the burning altar

ExcessiveUseOfCobblestone commented on code/modules/religion/rites.dm:
```diff
@@ -102,3 +102,118 @@
+		return ..()
+
+/datum/religion_rites/burning_sacrifice/invoke_effect(mob/living/user, atom/movable/religious_tool)
+	for(chosen_sacrifice in religious_tool.buckled_mobs) //checks one last time if the right corpse is still buckled
```
you dont need to loop here, if works.

that said you should check the inverse and early return.

Caldony committed:
change the for loop to an if check for the sacrifice

also added a bunch of early returns which forces me to set chosen sacrifice to null a whole lot more

Caldony committed:
updqte sect description
